pkgs/development/go-modules/generic/default.nix

```diff
@@ -198,7 +198,7 @@ go.stdenv.mkDerivation (
      ln -s "${dep.src}" "$d/src/${dep.goPackagePath}"
   ''
   ) goPath) + ''
-    export GOPATH="$d:$GOPATH"
+    export GOPATH=${lib.concatStringsSep ":" ( ["$d"] ++ ["$GOPATH"] ++ ["$PWD"] ++ extraSrcPaths)}
```

Comment:
```
(12:47) <  qknight_> if 'go build' yields: cannot find package "nixcloud/backend/root" in any of and my project is src/nixcloud/backends then i simply lack a entry in GOPATH with the absolute path to the 
                     parent of 'src', right?
(12:47) <  qknight_> because if i add this, it compiles
(12:47) -!- Poky [~michal@ip-89-177-75-53.net.upcbroadband.cz] has quit [Ping timeout: 268 seconds]
(12:48) <  qknight_> so that said, the project root must always be part of the GOPATH or is that wrong?
(12:48) -!- mperillo [~mperillo@host169-59-dynamic.43-79-r.retail.telecomitalia.it] has joined #go-nuts
(12:48) <      acln> qknight_: that is correct, all of your Go code usually lives under GOPATH.
```